src/snowflake/snowpark/modin/plugin/_internal/frame.py

```diff
@@ -609,16 +616,30 @@ def has_unique_index(self, axis: Optional[int] = 0) -> bool:
         if axis == 1:
             return self.data_columns_index.is_unique
         else:
-            # Note: We can't use 'count_distinct' because it ignores null values.
-            total_rows = self.num_rows
-            distinct_rows = count_rows(
-                get_distinct_rows(
-                    self.ordered_dataframe.select(
-                        self.index_column_snowflake_quoted_identifiers
+            # Implement using single query for single index.
+            if self.num_index_columns == 1:
```

Comment:
why can't we do the same for multiple index columns?

Reply:
For multiple index columns expression will be very complex. Happy to update the code if you have any suggestion for the expression.

Reply:
I think this works:

```python
            return self.ordered_dataframe._dataframe_ref.snowpark_dataframe.select(
                count_distinct(array_construct(*self.index_column_snowflake_quoted_identifiers)) == count("*"),
            ).collect()[0][0]
```

Reply:
interesting idea, let me try this.